### ISSUE TITLE
Support dns-01 challenges

### ISSUE THREAD

**Issue by mithrandi** [twisted/txacme#45]
This ticket is for tracking `dns-01` support in general; #32 covers one specific backend for dns-01 challenges.

This is blocked on certbot/certbot#2061 which adds support for the `dns-01` challenge to the `acme` library, which is used by `txacme` for all the protocol stuff. I'm not sure if any support code is needed in txacme beyond this and the `IResponder` interface, but at the very least, there should be documentation about how to implement your own responder (eg. inserting records into the SQL database backing a PowerDNS deployment).

**Comment by glyph** [comment]
relvant 2 my intrsts https://github.com/major/letsencrypt-rackspace-hook

**Comment by Lukasa** [comment]
I'm also registering my interest: I want to run a few Twisted's all serving the same content (because redundancy is good), but don't want to mint a new cert for each one of them. This would be a useful thing to have.

**Comment by glyph** [comment]
Should there maybe be a separate issue for "high availability certificate provisioning" since that probably involves a little more coordination beyond just the challenge?

**Comment by mithrandi** [comment]
Would that be a documentation issue? I don't think txacme can meaningfully tackle that beyond the extension points that it already provides (mainly `ICertificateStore`), but I can imagine having some prose and/or examples to give you some inspiration on how to assemble a solution for your particular environment. It's definitely out of scope for this issue, anyhow, so feel free to open another issue if you think you have some useful words to write on the subject.

**Comment by glyph** [comment]
So uh, I've been using this to provision all of my LAN certificates for like 6 months now. Should this be closed? What remains to do?

**Comment by mithrandi** [comment]
I was keeping this open to target a more general interface for implementing DNS challenge responders, but I think given what #59 looked like, there's not actually much need for an extra layer here, so I'm closing it.
